src/NHibernate.Test/NHSpecificTest/NH3023/DeadlockHelper.cs

```diff
@@ -57,6 +60,8 @@ public void ForceDeadlockOnConnection(SqlConnection connection)
 					if (winnerEx != null)
 						_log.Warn("Winner thread failed", winnerEx);
 				}
+				// If getting here, expected victim has not fail. If expected winner has failed instead, fail the test.
+				Assert.That(winnerEx, Is.Null);
```

Comment:
In case of winner thread failure while loser thread does not fail, the test was "succeeding". Prevent this.